src/pythoninlua.c

```diff
@@ -183,8 +184,23 @@ static int py_object_call(lua_State *L)
         ret = py_convert(L, value);
         Py_DECREF(value);
     } else {
-        PyErr_Print();
-        luaL_error(L, "error calling python function");
+        char s_err[1024];
+        memset(s_err, 0, 1024);
+
+        PyObject *exc_type, *exc_value, *exc_traceback;
+        PyErr_Fetch(&exc_type, &exc_value, &exc_traceback);
```

Comment:
The [documentation](https://docs.python.org/3/c-api/exceptions.html#c.PyErr_Fetch) says, we should use PyErr_GetRaisedException instead.

Reply:
As per the [documentation](https://docs.python.org/3/c-api/exceptions.html#c.PyErr_GetRaisedException) PyErr_GetRaisedException is available from 3.12. And 3.10 does not have it (I am using it and got an error) and earlier versions than 3.12 are very much in use.

PyErr_Fetch as such is deprecated and migration to the newer approach may have to be at a more suitable time.

Reply:
Have further improved the implementation in the second commit, where traceback.format_exception has been used to get complete information.

Reply:
I see, that's good then.